app/deploy.go

```diff
@@ -244,6 +239,14 @@ func Deploy(opts DeployOptions) (string, error) {
 	return imageId, nil
 }
 
+func RollbackUpdate(appName, imgId, reason string, rollback bool) error {
+	err := image.ValidateAppImage(appName, imgId)
```

Comment:
This will probably always fail. This happens because the imgId here most likely isn't a full image id, it's just the version suffix. The logic for transforming the received suffix into a full image id currently is [here in the Deploy method](https://github.com/tsuru/tsuru/pull/1561/files#diff-975107ae4993861af2c7808681346ecdR208). You should probably extract it to a function and use it in both places.